cmd/solvercli/cmd/balances.go

```diff
@@ -0,0 +1,245 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/skip-mev/go-fast-solver/shared/clients/skipgo"
+	"github.com/skip-mev/go-fast-solver/shared/config"
+	"github.com/skip-mev/go-fast-solver/shared/lmt"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+type SkipBalancesRequest struct {
+	Chains map[string]ChainRequest `json:"chains"`
+}
+
+type ChainRequest struct {
+	Address string   `json:"address"`
+	Denoms  []string `json:"denoms"`
+}
+
+type SkipBalancesResponse struct {
+	Chains map[string]ChainResponse `json:"chains"`
+}
+
+type ChainResponse struct {
+	Address string                 `json:"address"`
+	Denoms  map[string]DenomDetail `json:"denoms"`
+}
+
+type DenomDetail struct {
+	Amount          string `json:"amount"`
+	Decimals        uint8  `json:"decimals"`
+	FormattedAmount string `json:"formatted_amount"`
+	Price           string `json:"price"`
+	ValueUSD        string `json:"value_usd"`
+}
+
+type ChainBalance struct {
+	ChainID    string
+	AssetDenom string
+	Balance    *big.Int
+	Symbol     string
+	Decimals   uint8
+	PriceUSD   *big.Float
+	ValueUSD   *big.Float
+}
+
+type ChainGasBalance struct {
+	ChainID           string
+	Balance           *big.Int
+	Symbol            string
+	Decimals          uint8
+	WarningThreshold  *big.Int
+	CriticalThreshold *big.Int
+}
+
+var balancesCmd = &cobra.Command{
+	Use:   "balances",
+	Short: "Show current on-chain balances (USDC and gas tokens)",
+	Long: `Show current on-chain balances for USDC and gas tokens across all configured chains.
+    
+Example:
```

Comment:
the example should probably go in the `Example` field

Reply:
same for other commands as well

Reply:
done